Branch-SDK/Branch-SDK/Branch.m

```diff
@@ -129,16 +130,30 @@ + (void) load {
 + (void) openLog {
     // Initialize the log
     @synchronized (self) {
-        if (!bnc_logURL) {
+        if (bnc_logURL) {
+            BNCLogSetOutputToURLByteWrap(bnc_logURL, 102400);
+        } else {
             BNCLogInitialize();
             BNCLogSetDisplayLevel(BNCLogLevelAll);
             bnc_logURL = BNCURLForBranchDirectory();
             bnc_logURL = [[NSURL alloc] initWithString:@"Branch.log" relativeToURL:bnc_logURL];
             BNCLogSetOutputToURLByteWrap(bnc_logURL, 102400);
-            BNCLogSetDisplayLevel(BNCLogLevelWarning);
+            BNCLogSetDisplayLevel(BNCLogLevelWarning);  // Default
+
+            // Try loading from the Info.plist
+            NSString *logLevelString = [[NSBundle mainBundle] infoDictionary][@"BranchLogLevel"];
+            if ([logLevelString isKindOfClass:[NSString class]]) {
+                BNCLogLevel logLevel = BNBLogLevelFromString(logLevelString);
+                BNCLogSetDisplayLevel(logLevel);
+            }
+
+            // Try loading from user defaults
```

Comment:
I wonder about loading from user defaults. In particular, I'd expect the Info.plist to override the user defaults. If I make a change in the Info.plist, and then rerun in Xcode using this change, I believe the Info.plist change would be ignored. I'm not sure what value the user defaults add here. If you're setting it through the Info.plist, that will always be present. If you're setting it programmatically, that will also always be present. Is the concern that a logging URI like the one introduced here should persistently override the log level specified by the app? What if someone sends a debugging URI to a release build in production and forever turns on extremely verbose logging? Maybe it would be best if the debugging URI's effect were not persistent.

Reply:
We need to be able to turn logging on in a production app so that Integrations Engineering can debug partner issues without using Charles proxy.

If the Info.plist overrides the user defaults then turning on the logging via Branch Link won't work when debugging issues that involve links that open a non-running app.

I think the possibility of turning on logging by accident is so low it would have to be a malicious attack, and the only side effect is that log messages would appear in the iPhone user's console log.

Reply:
I see your point. No matter what workarounds you put in place for other builds, an AppStore release should have pinning enabled and logging disabled. This provides something.

I agree that the likelihood of this causing any serious problem is small.

:+1: